lib/internal/modules/esm/loader.js

```diff
@@ -806,9 +806,9 @@ class ModuleLoader {
    * if any.
    * @param {string} url The URL of the module to be loaded.
    * @param {object} context Metadata about the module
-   * @returns {Promise<{ format: ModuleFormat, source: ModuleSource }>}
+   * @returns {Promise<{ format: ModuleFormat, source: ModuleSource }> | { format: ModuleFormat, source: ModuleSource }}
```

Comment:
I think that this can be fully synchronous if it uses `this.#customizations.loadSync` below. But I am fine with leaving that to another PR.